femwell/maxwell/waveguide.py

```diff
@@ -148,6 +149,27 @@ def calculate_pertubated_neff(self, delta_epsilon):
             * 0.5
         )
 
+    def calculate_intensity(self, field: NDArray) -> Tuple[NDArray, Basis]:
+        """Calculates the intensity of a field as the sum of the absolute values squared of its components.
```

Comment:
Could you add the formula for the intensity (with a literature reference) to the docs? :)

Reply:
Sure!

Reply:
Sure!